public/jwpltx.js

```diff
@@ -37,10 +10,44 @@ window.jwpltx = window.jwpltx || {};
     oosv: '5',
     sdk: '0',
   };
+
+  // There are anywhere between 1 to 128 quantiles in any given video, 128 is max for every video
+  const MAX_DURATION_IN_QUANTILES = 128;
   // query params instance.
   let uri;
-  // Current time for live streams.
-  let current;
+  // Time watched after last t event was sent
+  let timeWatched = 0;
+  // Last progress of the video stored
+  let lastVp = 0;
+  // Last quantile reached
+  let lastQ;
+  // Seeking state, gets reset after 1 sec of the latest seeked event
+  let isSeeking = null;
+  // Timeout for seeking state
+  let liveInterval = null;
+
+  // How many quantiles have we passed (whether we need to send new t event or not)
+  function getLastVODQuantile(progress, duration, numberOfQuantiles) {
+    return Math.floor(progress / (duration / numberOfQuantiles));
+  }
```

Comment:
Let's not call this a quantile because the quantiles are 1-128 and this is 1-32 (max depending on the chunks.)
Also, suggest adding 1 so it matches the 1-32 instead of going 0-31.

```suggestion
  // How many quantiles have we passed (whether we need to send new t event or not)
  function getLastChunkNumber(progress, duration) {
    return Math.floor(progress / duration * uri.q) + 1;
  }
```